crypto/bn/bn_recp.c

```diff
@@ -63,6 +63,7 @@ void BN_RECP_CTX_init(BN_RECP_CTX *recp)
     bn_init(&(recp->N));
     bn_init(&(recp->Nr));
     recp->num_bits = 0;
+    recp->shift = 0;
```

Comment:
As the two methods are public, they deserve both to be fixed , isn't it ?
I push-ed a separate commit about the missing init, to ease its merge.